src/kaleidoscope/key_defs.h

```diff
@@ -202,6 +203,45 @@ typedef kaleidoscope::Key Key_;
 #define ___ Key_Transparent
 #define XXX Key_NoKey
 
+namespace kaleidoscope {
+constexpr bool isKeyboardKey(Key key) {
+  return ((key.getFlags() & (RESERVED | SYNTHETIC)) == 0);
+}
+constexpr bool isKeyboardModifier(Key key) {
+  return (isKeyboardKey(key) &&
+          (key.getKeyCode() >= HID_KEYBOARD_FIRST_MODIFIER &&
+           key.getKeyCode() <= HID_KEYBOARD_LAST_MODIFIER));
+}
+constexpr bool isKeyboardShift(Key key) {
+  return (isKeyboardModifier(key) &&
+          ((key.getKeyCode() == HID_KEYBOARD_LEFT_SHIFT ||
+            key.getKeyCode() == HID_KEYBOARD_RIGHT_SHIFT) ||
+           ((key.getFlags() & SHIFT_HELD) != 0)));
+}
+
+constexpr bool isLayerKey(Key key) {
```

Comment:
I don't have a great naming proposal for "isLayerKey" - I think I'd like it to not end in the word "Key"

Reply:
I'm using the `Key` postfix for the functions that are determining which variant of the `Key` class the object is. Plugins can then get in on this game: `isOneShotKey()`, `isTapDanceKey()`, `isDualUseKey`, _et cetera_.